pkg/workspace/data_values_pre_processing.go

```diff
@@ -37,46 +38,89 @@ func (o DataValuesPreProcessing) Apply() (*DataValues, []*DataValues, error) {
 }
 
 func (o DataValuesPreProcessing) apply(files []*FileInLibrary) (*DataValues, []*DataValues, error) {
-	values := o.loader.schema.AsDataValue()
-	var libraryValues []*DataValues
-	for _, fileInLib := range files {
-		valuesDocs, err := o.templateFile(fileInLib)
-		if err != nil {
-			return nil, nil, fmt.Errorf("Templating file '%s': %s", fileInLib.File.RelativePath(), err)
-		}
+	_, checkSchema := o.loader.schema.(*schema.DocumentSchema)
 
-		for _, valuesDoc := range valuesDocs {
-			dv, err := NewDataValues(valuesDoc)
+	allDvs, err := o.collectDataValuesDocs(files)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// merge all Data Values YAML documents into one
+	var dvsForOtherLibraries []*DataValues
+	var dataValuesDoc *yamlmeta.Document
+	for _, dv := range allDvs {
+		switch {
+		case dv.HasLibRef():
+			dvsForOtherLibraries = append(dvsForOtherLibraries, dv)
+		case dataValuesDoc == nil:
+			err := o.loader.schema.ValidateWithValues(1)
 			if err != nil {
 				return nil, nil, err
 			}
 
-			switch {
-			case dv.HasLib():
-				libraryValues = append(libraryValues, dv)
-			case values == nil:
-				values = valuesDoc
-			default:
-				var err error
-				values, err = o.overlay(values, dv.Doc)
-				if err != nil {
-					return nil, nil, err
-				}
+			dataValuesDoc = dv.Doc
+		default:
+			if checkSchema {
+				o.annotateWithDefaultMissingOk(dv.Doc)
```

Comment:
let's pull this out as a separate PR. by adding missing_ok=True in certain cases we are affecting how overlay feature behaves under certain conditions. that's not desirable because we want users to know that same piece of code behaves same way regardless where it's being called from.

quick example that comes to mind: 
```
#@overlay/match by=overlay.subset({"name":"foo"})
- blah: something
```
will behave differently since since overlay module by default sets expects=1.

to start resolving this situtation, i think we need to figure out what happens in various interactions between schema and overlay. what errors would one expect to see when new keys are used, what errors do we expect to see when new keys are used with any type, etc.. let's have a document enumerating these cases and discuss them.

Reply:
... and to clarify, it's not so much that we'd pull this out as a separate PR... but that we'd _either_:

- address this problem _within_ this PR, or,
- temporarily accept this intermediate state where it generally works but has undesired side-effects in some cases... but then _immediately_ address this problem in a subsequent PR.  (consider this change set "large enough" and address this as a separate unit of work)

But either way, we can't remove the `--enable-experiment-schemas` flag until we ensure that this works exact the way we want it.

Reply:
We pulled out this to a separate PR where we plan to address this. The PR is #337 and is a prerequisite to this work so that PR must be merged in prior to this one.